test/e2e/framework/pod/resource.go

```diff
@@ -205,6 +206,17 @@ func podRunning(c clientset.Interface, podName, namespace string) wait.Condition
 		}
 		switch pod.Status.Phase {
 		case v1.PodRunning:
+			// Verify that all containers are running
+			for _, ctr := range pod.Status.ContainerStatuses {
+				state := ctr.State
+				if state.Waiting != nil && state.Waiting.Reason == kubecontainer.ErrRunContainer.Error() {
```

Comment:
even if `state.Waiting` is non-nil, `state.Running` is what we really care about as per the function name.
can we get way with only checking `if state.Running == nil`?

also how about containers that have finished without error?

Reply:
Hm yeah I added this branch because returning an error will make the test fail earlier and not "only timeout". But I think we can just check for `Running == nil`, sure.